Reorganising the documentation to decrease scatter

Diogenesoftoronto opened the pull request:
There is now a docs folder and the top level README includes links to
other files.

Closes #512

Diogenesoftoronto committed:
Reorganise documentation

There is now a docs folder and the top level README includes links to
other files.

djjuhasz commented on docs/deps.md:
```diff
@@ -0,0 +1,84 @@
+
+### `goa.design/goa/v3`
+
+Update the dependency:
```
Goa now has a `hack/make/dep_goa.mk` file like ent does, so it should now be updated in the same way as ent.

djjuhasz commented on docs/release.md:
```diff
@@ -0,0 +1,18 @@
+# Release
```
Enduro doesn't have a `VERSION.txt` file or versioning system yet, so I think this file should be deleted for now.

Diogenesoftoronto replied:
BTW on the release stuff it might make sense to suggest using github's release system. I can even add something like that in the docs with a guide for how to use gh cli to create release versions. Secondly, this repo lacks a contributor.md file this might be a good idea.

djjuhasz replied:
I don't know anything about Github's release system, so I can't comment on whether we should use it for Enduro.

I agree that a `contributor.md` would be a good addition.

Diogenesoftoronto replied:
I will use a temporary contributor md and then add an issue to work on it later

djjuhasz reviewed (changes requested):
@Diogenesoftoronto looks good overall, just a couple of change requests.

Diogenesoftoronto committed:
Remove release.md and add dep goa update

djjuhasz commented on docs/devel.md:
```diff
@@ -0,0 +1,252 @@
+is sometimes not setup properly. To solve it, from the Tilt UI, restart the
+`minio` resource and then trigger the `minio-setup-buckets` resource.
+
+[national health archive (nha)]: https://www.piql.com/norwegians-digital-health-data-to-be-preserved-for-future-generations/
```
I don't think this link is used in the docs and should be removed.

djjuhasz commented on docs/devel.md:
```diff
@@ -0,0 +1,252 @@
+`minio` resource and then trigger the `minio-setup-buckets` resource.
+
+[national health archive (nha)]: https://www.piql.com/norwegians-digital-health-data-to-be-preserved-for-future-generations/
+[national center for truth and reconciliation (nctr)]: https://nctr.ca/about/about-the-nctr/our-mandate/
```
I don't think this link is used in the docs and should be removed.

djjuhasz reviewed (changes requested):
@Diogenesoftoronto I found a couple of links that I think are old and should be removed.

Diogenesoftoronto committed:
Remove dead links

Diogenesoftoronto committed:
Add a contributor file

A temporary contributor file has been added that will be expanded on in
a future PR

Diogenesoftoronto committed:
Add necessary links

djjuhasz commented on CONTRIBUTOR.md:
```diff
@@ -0,0 +1,3 @@
+# Contributing
+
+We welcome contributions from everyone. More details on how to contribute will be coming. For now simply submit a pull-request or create an issue.
```
@Diogenesoftoronto we can't actually accept contributions without a signed contributors agreement (See https://github.com/artefactual/archivematica/blob/stable/1.14.x/CONTRIBUTING.md#contributors-agreement for an example).  I think it's better to omit the contributor file for now, and add one later when we have worked out the contents.

Diogenesoftoronto replied:
Sensible

Diogenesoftoronto committed:
Remove the contributor document

djjuhasz reviewed (approved):
LGTM @Diogenesoftoronto.  Don't forget to squash the commits before merging to main.
